### PR TITLE
Add support for glTF normalTextureInfo.scale

### PR DESCRIPTION
**Pull request by jjhembd**
# Description

The [glTF specification](https://registry.khronos.org/glTF/specs/2.0/glTF-2.0.html#_material_normaltextureinfo_scale) describes a scale that can be applied to the values read from a normal texture. This scalar applies to the `.x` and `.y` components only.

This PR adds a `.scale` property to `ModelComponents.TextureReader`, and adds code to process the scale in `GltfLoaderUtil` and `MaterialPipelineStage`. In the shader, the scalar is applied (if present) in `MaterialStageFS`.

Here is the Clearcoat Car Paint test model in CesiumJS 1.118:
image

With this PR:
image

Comparing to the [glTF Sample Viewer](https://github.khronos.org/glTF-Sample-Viewer-Release/)
image

## Issue number and link

Resolves https://github.com/CesiumGS/cesium/issues/6948.

## Testing plan

Load the [glTF PBR Extensions Sandcastle](http://localhost:8080/Apps/Sandcastle/index.html?src=glTF%20PBR%20Extensions.html) and compare the Clearcoat Car Paint model to the rendering in the [glTF Sample Viewer](https://github.khronos.org/glTF-Sample-Viewer-Release/)

# Author checklist

- [x] I have submitted a Contributor License Agreement
- [x] I have added my name to `CONTRIBUTORS.md`
- [x] I have updated `CHANGES.md` with a short summary of my change
- [x] I have added or updated unit tests to ensure consistent code coverage
- [x] I have update the inline documentation, and included code examples where relevant
- [x] I have performed a self-review of my code

### CODE REVIEW

File: packages/engine/Source/Scene/Model/MaterialPipelineStage.js [hunk 1]
```diff
@@ -216,6 +216,44 @@ function processTextureTransform(
   };
 }
 
+/**
+ * Process a single texture scale and add it to the shader and uniform map.
+ *
+ * @param {ShaderBuilder} shaderBuilder The shader builder to modify
+ * @param {Object<string, Function>} uniformMap The uniform map to modify.
```
Comment (ggetz) [inline packages/engine/Source/Scene/Model/MaterialPipelineStage.js:223]:
```suggestion
 * @param {Object<string, function>} uniformMap The uniform map to modify.
```

**Review by ggetz — COMMENTED** [review]
Looks good to me @jjhembd! Should we add or update any unit tests?

File: packages/engine/Source/Scene/Model/MaterialPipelineStage.js (hunk 1 above)
Comment (jjhembd) [reply packages/engine/Source/Scene/Model/MaterialPipelineStage.js:223]: I see your point here, we normally don't use the capitalized names for primitive types. However, in the case of `function`, I think we are capitalized everywhere. Apparently this wasn't addressed in https://github.com/CesiumGS/cesium/pull/11080.
Do we want to start fixing these? Or wait for https://github.com/CesiumGS/cesium/issues/10815?

Comment (ggetz) [reply packages/engine/Source/Scene/Model/MaterialPipelineStage.js:223]: Gotcha, my bad. We can hold off until #10815.

**Comment by jjhembd** [comment]
Thanks @ggetz, I added/updated some specs. This is ready for another look.

**Comment by ggetz** [comment]
Awesome, thanks @jjhembd!

**Pull request merged**